arcgis-ios-sdk-samples/Layers/Map image layer tables/MapImageLayerTablesViewController.swift

```diff
@@ -0,0 +1,158 @@
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class MapImageLayerTablesViewController: UIViewController {
+    @IBOutlet var mapView: AGSMapView! {
+        didSet {
+            // Assign the map to the map view.
+            mapView.map = makeMap()
+            // Set the viewpoint.
+            mapView.setViewpoint(AGSViewpoint(latitude: 41.734152, longitude: -88.163718, scale: 2e5))
+            // Add a graphics overlay to show selected features and add it to the map view.
+            selectedFeaturesOverlay.renderer = makeRenderer()
+            mapView.graphicsOverlays.add(selectedFeaturesOverlay)
+        }
+    }
+    
+    @IBOutlet var queryBarButtonItem: UIBarButtonItem!
+    
+    /// The service request URL.
+    static let serviceRequestURL = URL(string: "https://sampleserver6.arcgisonline.com/arcgis/rest/services/ServiceRequest/MapServer")!
+    /// The map image layer that uses the service URL.
+    let serviceRequestsMapImageLayer = AGSArcGISMapImageLayer(url: serviceRequestURL)
+    /// The (non-spatial) table that contains the service request comments.
+    var commentsTable: AGSServiceFeatureTable?
+    /// The array to store the possible comments.
+    var commentsArray: [AGSFeature] = []
+    /// The graphics overlay to add graphics to.
+    let selectedFeaturesOverlay = AGSGraphicsOverlay()
+    
+    @IBAction func queryFeaturesActions(_ sender: UIBarButtonItem) {
+        // Create an action sheet to display the various comments to choose from.
+        let alertController = UIAlertController(title: "Related Service Requests", message: "Select a comment to view related spatial features on the map.", preferredStyle: .actionSheet)
+        // Create an action for each comment.
+        commentsArray.forEach { feature in
+            // Extract the "comments" attribute as a string.
+            let commentsTitle = feature.attributes["comments"] as! String
+            // Create an action with the comments title.
+            let action = UIAlertAction(title: commentsTitle, style: .default) { [weak self] (_) in
+                guard let self = self else { return }
+                // Clear the former graphics.
+                self.selectedFeaturesOverlay.graphics.removeAllObjects()
+                // Disable the query button while the feature loads.
+                self.queryBarButtonItem.isEnabled = false
+                // Cast the selected feature as an AGSArcGISFeature.
+                guard let selectedFeature = feature as? AGSArcGISFeature else { return }
```

Comment:
IMO the `selectedFeature` is always an `AGSArcGISFeature`, as `commentsArray` is guaranteed to be an array of `AGSArcGISFeature`. So you can force unwrap here. 🤔

Reply:
`commentsArray` is actually type `[AGSFeature]`. Do you still think I should force unwrap?

Reply:
I'm not too sure. I'll need to talk to others to understand the difference between `AGSFeature` and `AGSArcGISFeature`. 🧐 
I made this suggestion because all options in the action sheet work, so the cast always succeed in this sample... 😅

Reply:
Today I had a brief chat with Divesh and he pointed out a few things, and I read the [doc](https://developers.arcgis.com/ios/api-reference/interface_a_g_s_arc_g_i_s_feature.html#:~:text=An%20AGSArcGISFeature%20object,entity%20it%20represents) again and believe that casting to `AGSArcGISFeature` should be fine for a map service.